csharp/method.go

```diff
@@ -73,6 +92,11 @@ func (self *MethodDeclaration) WithBase(args ...string) *MethodDeclaration {
 	return self
 }
 
+func (self *MethodDeclaration) Summary(summary string) *MethodDeclaration {
```

Comment:
You might want to note that ordering is important wrt this call.

Reply:
I've changed the logic a little bit, so we can add params and returnType before the summary itself and it'll still write the summary comments.